src/pages/ReportAddApproverPage.tsx

```diff
@@ -0,0 +1,205 @@
+
+    const employeeList = policy?.employeeList;
+    const sections = useMemo(() => {
+        const approvers: SelectionListApprover[] = [];
+
+        if (employeeList) {
+            const policyMemberEmailsToAccountIDs = getMemberAccountIDsForWorkspace(employeeList);
+            const availableApprovers = Object.values(employeeList)
+                .map((employee): SelectionListApprover | null => {
+                    const isAdmin = employee?.role === CONST.REPORT.ROLE.ADMIN;
+                    const email = employee.email;
+
+                    if (!email) {
+                        return null;
+                    }
+                    const accountID = Number(policyMemberEmailsToAccountIDs[email] ?? CONST.DEFAULT_NUMBER_ID);
+
+                    if (report.managerID === accountID) {
+                        return null;
+                    }
+
+                    const {avatar} = personalDetails?.[accountID] ?? {};
+                    const displayName = getDisplayNameForParticipant({accountID, personalDetailsData: personalDetails});
+                    return {
+                        text: displayName,
+                        alternateText: email,
+                        keyForList: email,
+                        isSelected: selectedApproverEmail === email,
+                        login: email,
+                        value: accountID.toString(),
+                        icons: [{source: avatar ?? FallbackAvatar, type: CONST.ICON_TYPE_AVATAR, name: displayName, id: accountID}],
+                        rightElement: isAdmin ? <Badge text={translate('common.admin')} /> : undefined,
+                    };
+                })
+                .filter((approver): approver is SelectionListApprover => !!approver);
+
+            approvers.push(...availableApprovers);
+            // eslint-disable-next-line react-compiler/react-compiler
+            setAllApprovers(approvers);
+        }
+
+        const filteredApprovers =
+            debouncedSearchTerm !== ''
+                ? tokenizedSearch(approvers, getSearchValueForPhoneOrEmail(debouncedSearchTerm, countryCode), (option) => [option.text ?? '', option.login ?? ''])
+                : approvers;
+
+        const data = sortAlphabetically(filteredApprovers, 'text', localeCompare);
+        return [
+            {
+                title: undefined,
+                data,
+                shouldShow: true,
+            },
+        ];
+    }, [employeeList, debouncedSearchTerm, countryCode, localeCompare, report.managerID, personalDetails, selectedApproverEmail, translate]);
+
+    const shouldShowListEmptyContent = !debouncedSearchTerm && !sections.at(0)?.data.length;
+
+    const addApprover = useCallback(() => {
+        const employeeAccountID = allApprovers.find((approver) => approver.login === selectedApproverEmail)?.value;
+        if (!selectedApproverEmail || !employeeAccountID) {
```

Comment:
I don't understand how either one of these things would happen. This would be a dead click without user feedback?